src/Entity/SmsDeliveryReport.php

```diff
@@ -178,35 +180,35 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
 
     $fields['status'] = BaseFieldDefinition::create('string')
       ->setLabel(t('Delivery status'))
-      ->setDescription(t('The status of the message. One of queued, delivered, expired, rejected, invalid_recipient or content_invalid.'))
+      ->setDescription(t('A status code from \Drupal\sms\Message\SmsMessageReportStatus.'))
       ->setReadOnly(TRUE)
       ->setRequired(TRUE)
       ->setRevisionable(TRUE);
 
     $fields['status_message'] = BaseFieldDefinition::create('string')
       ->setLabel(t('Status message'))
-      ->setDescription(t('The status message describing the message status.'))
+      ->setDescription(t('The status message as provided by the gateway API.'))
       ->setReadOnly(TRUE)
       ->setDefaultValue('')
       ->setRequired(FALSE)
       ->setRevisionable(TRUE);
 
-    $fields['status_time'] = BaseFieldDefinition::create('string')
+    $fields['status_time'] = BaseFieldDefinition::create('timestamp')
       ->setLabel(t('Status time'))
-      ->setDescription(t('The time specified by the server for the current status.'))
+      ->setDescription(t('The time for the current delivery report status.'))
       ->setReadOnly(TRUE)
       ->setRequired(TRUE)
       ->setRevisionable(TRUE);
 
     $fields['changed'] = BaseFieldDefinition::create('changed')
       ->setLabel(t('Changed'))
-      ->setDescription(t('The time that the delivery report was last changed.'))
+      ->setDescription(t('The time the storage was last updated.'))
```

Comment:
Well, changed refers to the time the delivery report storage was updated. Not when the delivery report itself was updated. Any better suggestions?